### PR TITLE
Replace client sdk with gateway client transport

### PR DESCRIPTION
**Pull request by segabriel**
The aim is to be able to use services that work in cluster with gateway in a simplified way. Once Services and Gateway are started, it will be enough to create ServiceCall that knows gateway's address to start consuming services.

That's so called gateway client transport. It can be of any supported by GW protocol: http/websocket/rsocket. Client transport is injected into service call, so there is no need in neither router nor discovery components. all that is done by gateway. That transport is replacing Client SDK

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@
     <module>services-gateway-netty</module>
     <module>services-gateway-tests</module>
     <module>services-gateway-runner</module>
-    <module>services-gateway-clientsdk</module>
+    <module>services-gateway-clientTransport</module>
```
Comment (ronenhamias) [inline pom.xml]: this should be named services-gateway-client-transport

File: services-gateway-benchmarks/src/main/java/io/scalecube/services/benchmarks/gateway/InfiniteStreamScenario.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ private static Integer rateLimit(BenchmarkSettings settings) {
         .orElse(DEFAULT_RATE_LIMIT);
   }
 
-  private static ClientMessage enichResponse(ClientMessage msg) {
-    return ClientMessage.from(msg).header(CLIENT_RECV_TIME, System.currentTimeMillis()).build();
+  private static ServiceMessage enichResponse(ServiceMessage msg) {
```
Comment (ronenhamias) [inline services-gateway-benchmarks/src/main/java/io/scalecube/services/benchmarks/gateway/InfiniteStreamScenario.java]: threr is typo here `enrich`

**Review by snripa — APPROVED** [review]

**Comment by dmytro-lazebnyi** [comment]
I found the following example of usage in `AbstractGatewayExtension`:

```
    Address address = this.gateway.gateway(gatewayId).address();
    client = Microservices.builder()
        .transport(op -> clientSupplier.apply(
            GwClientSettings.builder().address(address).loopResources(clientLoopResources).build(),
            op))
        .startAwait();

    serviceCall = client.call().router(new StaticAddressRouter(address));
```

So, if I understood correctly, now to call gateway I need to do following steps on a ***caller side***:
1. Add dependency to whole `scalecube-services` module.
2. Build and start new `Microservices`.
3. Create `ServiceCall`.

Looks too overhead as for me.

**Comment by ronenhamias** [comment]
> I found the following example of usage in `AbstractGatewayExtension`:
> 
> ```
>     Address address = this.gateway.gateway(gatewayId).address();
>     client = Microservices.builder()
>         .transport(op -> clientSupplier.apply(
>             GwClientSettings.builder().address(address).loopResources(clientLoopResources).build(),
>             op))
>         .startAwait();
> 
>     serviceCall = client.call().router(new StaticAddressRouter(address));
> ```
> 
> So, if I understood correctly, now to call gateway I need to do following steps on a _**caller side**_:
> 
> 1. Add dependency to whole `scalecube-services` module.
> 2. Build and start new `Microservices`.
> 3. Create `ServiceCall`.
> 
> Looks too overhead as for me.

1. there is no need for microservices. i have discussed this with ripa and we agreed its not required.
2. if you have microservices so you can use it so calling from microservices should be optional.
3. if you want to call gw from existing service so now it makes more sense instead of adding a complete duplication of existing functionality (seem like overhead to me)
4. if you just want to call GW so you add reference to services as you would for client sdk (wired name).

in short: client SDK is a duplication to ServiceCall and requires its maintenance and releases . adding this dependency makes client sdk evolution slow and unnatural.

**Review by ronenhamias — CHANGES_REQUESTED** [review]
client SDK is completely redundant

the changes should be made on service call.

it should be possible to instantiate a service call without router directlly to GW address.

```java
if(address!=null){ // making static call to api gateway
   return requestOne(request, responseType, address));
} else if (methodRegistry.containsInvoker(qualifier)) { // local service.
```  

```java
GreetingService service = ServiceCall.call().transport(<GatewayTransport>).address(address).api(GreetingService.class);
```

**Review by artem-v — CHANGES_REQUESTED** [review]
A `io.scalecube.services.ServiceCall` depends already on `scalecube-api` (in majority of imports). Only 
```
import io.scalecube.services.routing.RoundRobinServiceRouter;
import io.scalecube.services.routing.Routers;
```
tights us to `scalecube-services`.

It looks like `ServiceCall` can live in module `api` and if one needs talk to gateway he picks up gateway-client-transport and that's his dependencies.

**Review by artem-v — APPROVED** [review]
